llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -2117,13 +2102,24 @@ SDValue SelectionDAG::getVScale(const SDLoc &DL, EVT VT, APInt MulImm,
   return getNode(ISD::VSCALE, DL, VT, getConstant(MulImm, DL, VT));
 }
 
+template <typename Ty>
```

Comment:
Can you add documentation for this function to describe what `VT` and `X` mean and how they are used to create the end result?